src/components/Layout/Layout.js

```diff
@@ -35,13 +37,21 @@ class Layout extends React.Component {
 
     if (this.props.search && this.props.search.display) {
       const values = queryString.parse(this.props.location.search);
+
       if (values.q) {
         this.toggleSearch(true);
         this.setState({ searchValue: values.q });
       } else {
         this.toggleSearch(false);
         this.setState({ searchValue: '' });
       }
+      const fromUrl = this.props.location.state.fromUrl || '/';
```

Comment:
Can this be undefined? If no then the next line will always be true?

Reply:
yep it can be 'undefined'